crates/forge/src/test_execution_syscall_handler.rs

```diff
@@ -698,29 +677,58 @@ fn execute_call_contract(
     let entry_point_selector = buffer.next_felt252().unwrap().into_owned();
 
     let calldata = buffer.next_arr().unwrap();
-
-    let call_result = call_contract(
-        blockifier_state,
-        cheatnet_state,
-        &contract_address,
-        &entry_point_selector,
-        &calldata,
+    cheatable_syscall_handler.syscall_handler.syscall_ptr += (buffer.ptr - system_ptr).unwrap();
```

Comment:
This is a confusing place to increment the syscall pointer

Reply:
Discussed offline, we don't have any idea to improve upon it